_drafts/screenshotter.md

```diff
@@ -0,0 +1,41 @@
+
+I was pretty happy with what I had done so far, but still, maximum automation this was not. See, there was still the matter of cropping the screenshots to a smaller size. A lot of the dialogs in Factory Planner are relatively small, and the having game world in the background would be distracting. I could do the cropping part manually, but I didn't want to. *Something something* lazyness. I had a potential solution in the back of my mind, but it would require some dirty work.
+
+The basic idea was to, as the scenario was taking the screenshots, also record the sizes of the dialogs whose picture was being taken. This metadata could then be written to a file as well, and the Python script could use it to crop the screenshots to the right size. This sounds like a great idea in theory, but it runs into a tiny issue in reality: The game doesn't let you read the dimensions of GUI windows, only write them.
+
+This is forbidden for good reason, as these dimensions are not actually deterministic in all cases, as a different game language for example might change the size of certain elements. Reading and using these undeterministic values in multiplayer would lead to desyncs, making the mod multiplayer-incompatible, and break replays too. Seemed like a dead end. Was I fated to forever manually crop screenshots? Turns out, I was not.
+
+Now, I couldn't solve the issue that reading frame dimensions leads to desyncs, that's just how the game works on a fundamental level. I am however not running the game in multiplayer to take these screenshots, and as it turns out, you can already launch the game in a special mode called [Instrument Mode](https://lua-api.factorio.com/latest/Instrument.html) that disables multiplayer and enables some unsafe functionality. Seems like the perfect place to slip in a super-secret hidden method that gives me access to the forbidden knowledge of frame dimensions. So I went ahead and put it in. Working for Wube does have its perks as it turns out.
+
+After several days of piling on the hacks, the beast finally roared to life, fully automating the process of taking screenshots for me. Although, *fully* automating the process is not quite correct. Turns out, it's still not fully automated. Of course it's not. Why would I get to be *fully* satisfied. There's two steps that I still need to do myself.
```

Comment:
```suggestion
After several days of piling on the hacks, the beast finally roared to life, fully automating the process of taking screenshots for me. Although *fully* automating the process is not quite correct. Turns out, it's still not fully automated. Of course it's not. Why would I get to be *fully* satisfied. There's two steps that I still need to do myself.
```

Reply:
I don't think `Turns out` should be replaced with `It turns out` here, but it depends on the balance of casual/formal you're aiming for. Although I've been changing it to be more formal in the previous suggested changes, you might want to make it consistently either casual or formal everywhere.